sqlmesh/dbt/model.py

```diff
@@ -119,6 +119,14 @@ def _validate_partition_by(cls, v: t.Any) -> t.Union[t.List[str], t.Dict[str, t.
         },
     }
 
+    @property
+    def time_column(self) -> t.Optional[str]:
+        if self.time_column_:
+            return self.time_column_
+        if isinstance(self.partition_by, dict) and self.partition_by["data_type"] != "int64":
```

Comment:
Wouldn't this fail if an engine other than BQ supports partition_by input as dictionary?

Reply:
No other engine does.